src/daemon/control.rs

```diff
@@ -614,784 +604,35 @@ fn announce_spend_transaction(
-                }
-                let mut finalized_spend = spend_tx.psbt.clone();
-                match finalized_spend.finalize(&revaultd.secp_ctx) {
-                    Ok(()) => {}
-                    Err(e) => {
-                        response_tx.send(Err(RpcControlError::CosigningServer(format!(
-                            "Invalid signature given by the cosigners, psbt: '{}' (error: '{}')",
-                            spend_tx.psbt.as_psbt_string(),
-                            e
-                        ))))?;
-                        continue;
-                    }
-                }
-
-                // And then announce it to the Coordinator
-                let deposit_outpoints = spent_vaults
-                    .values()
-                    .map(|db_vault| db_vault.deposit_outpoint)
-                    .collect();
-                match announce_spend_transaction(&revaultd, finalized_spend, deposit_outpoints) {
-                    Ok(()) => {}
-                    Err(e) => {
-                        response_tx.send(Err(RpcControlError::Communication(e.to_string())))?;
-                        continue;
-                    }
-                }
-                db_update_spend(&db_path, &spend_tx.psbt)?;
-
-                // Finally we can broadcast the Unvault(s) transaction(s) and store the Spend
-                // transaction for later broadcast
-                match bitcoind_broadcast_unvaults(
-                    &bitcoind_tx,
-                    &db_path,
-                    &revaultd.secp_ctx,
-                    &spent_vaults,
-                ) {
-                    Ok(()) => {}
-                    Err(e) => {
-                        response_tx.send(Err(RpcControlError::UnvaultBroadcast(e.to_string())))?;
-                        continue;
-                    }
-                }
-                db_mark_broadcastable_spend(&db_path, &spend_txid)?;
+#[derive(Clone)]
+pub struct RpcUtils {
+    pub revaultd: Arc<RwLock<RevaultD>>,
+    pub db_path: PathBuf,
+    pub network: Network,
+    pub bitcoind_tx: Sender<BitcoindMessageOut>,
+    pub bitcoind_thread: Arc<RwLock<Option<JoinHandle<()>>>>,
+    pub sigfetcher_tx: Sender<SigFetcherMessageOut>,
+    pub sigfetcher_thread: Arc<RwLock<Option<JoinHandle<()>>>>,
+}
 
-                response_tx.send(Ok(()))?;
-            }
+impl RpcUtils {
+    pub fn new(
+        revaultd: Arc<RwLock<RevaultD>>,
+        db_path: PathBuf,
+        network: Network,
+        bitcoind_tx: Sender<BitcoindMessageOut>,
+        bitcoind_thread: JoinHandle<()>,
+        sigfetcher_tx: Sender<SigFetcherMessageOut>,
+        sigfetcher_thread: JoinHandle<()>,
+    ) -> Self {
+        Self {
+            revaultd,
+            db_path,
+            network,
+            bitcoind_tx,
+            bitcoind_thread: Arc::new(RwLock::new(Some(bitcoind_thread))),
+            sigfetcher_tx,
+            sigfetcher_thread: Arc::new(RwLock::new(Some(sigfetcher_thread))),
```

Comment:
Why having a constructor if fields are public ?

Reply:
I honestly don't know what the convention is, but using constructors only when fields are private seems reasonable. I'll update it :)